### PR TITLE
24: Tasks listing.

### PR DESCRIPTION
**Pull request by robherba**
**Steps to test:**
- Create new project node.
- Create multiple tasks for testing and associates these to the new project.
- Check if the tasks are added in the project detaille.

**DoD**
- Go to a project.
- It should have a content pane view that lists all associated tasks (display should use panels).

[issue link](https://github.com/ManatiCR/manatime/issues/24)

### CODE REVIEW

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 1]
```diff
@@ -1,26 +1,33 @@
 name = Manatime tasks feature
 core = 7.x
 package = manatime
-version = 7.x-1.0
-dependencies[] = ctools
-dependencies[] = entityreference
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info:6]: Add this dependency back please.

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 2]
```diff
@@ -1,26 +1,33 @@
...
-dependencies[] = features
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info:7]: Add this dependency back please.

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 3]
```diff
@@ -1,26 +1,33 @@
...
 features[variable][] = node_preview_tasks
 features[variable][] = node_submitted_tasks
-project path = sites/all/modules/features
+features[views_view][] = tasks_listing
+features_exclude[dependencies][ctools] = ctools
+features_exclude[dependencies][entityreference] = entityreference
+features_exclude[dependencies][features] = features
+features_exclude[dependencies][list] = list
+features_exclude[dependencies][options] = options
+project path = sites/all/modules
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info]: sites/all/modules/features

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 4]
```diff
@@ -1,26 +1,33 @@
...
 features[variable][] = field_bundle_settings_node__tasks
 features[variable][] = menu_options_tasks
 features[variable][] = menu_parent_tasks
 features[variable][] = node_options_tasks
 features[variable][] = node_preview_tasks
 features[variable][] = node_submitted_tasks
-project path = sites/all/modules/features
+features[views_view][] = tasks_listing
+features_exclude[dependencies][ctools] = ctools
+features_exclude[dependencies][entityreference] = entityreference
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info]: Delete

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 5]
```diff
@@ -1,26 +1,33 @@
...
+features[page_manager_handlers][] = node_view__panel_context_021763f9-9425-45b7-951f-acb6b5b39ada
 features[variable][] = field_bundle_settings_node__tasks
 features[variable][] = menu_options_tasks
 features[variable][] = menu_parent_tasks
 features[variable][] = node_options_tasks
 features[variable][] = node_preview_tasks
 features[variable][] = node_submitted_tasks
-project path = sites/all/modules/features
+features[views_view][] = tasks_listing
+features_exclude[dependencies][ctools] = ctools
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info]: Delete

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 6]
```diff
@@ -1,26 +1,33 @@
...
+features_exclude[dependencies][entityreference] = entityreference
+features_exclude[dependencies][features] = features
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info]: Delete

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 7]
```diff
@@ -1,26 +1,33 @@
...
+features_exclude[dependencies][list] = list
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info]: Delete

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info [hunk 8]
```diff
@@ -1,26 +1,33 @@
...
+features_exclude[dependencies][options] = options
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.info]: Delete

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc [hunk 9]
```diff
@@ -0,0 +1,126 @@
...
+  $handler->display->display_options['filters']['type']['field'] = 'type';
+  $handler->display->display_options['filters']['type']['value'] = array(
+    'tasks' => 'tasks',
+  );
+
+  /* Display: Content pane */
+  $handler = $view->new_display('panel_pane', 'Content pane', 'panel_pane_1');
+  $handler->display->display_options['argument_input'] = array(
+    'nid' => array(
+      'type' => 'user',
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc:117]: I think this is not the right input.
Look for "Node:ID" or something like that.

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc [hunk 10]
```diff
@@ -0,0 +1,126 @@
...
+    'field_billable' => array(
+      'sortable' => 0,
+      'default_sort_order' => 'asc',
+      'align' => '',
+      'separator' => '',
+      'empty_column' => 0,
+    ),
+  );
+  /* Relationship: Entity Reference: Referenced Entity */
+  $handler->display->display_options['relationships']['field_project_target_id']['id'] = 'field_project_target_id';
```
Comment (kporras07) [inline modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc:55]: I haven't checked this; are you sure this is the right relationship? I mean; if you add this to the page; does it get the right tasks?

**Comment by kporras07** [comment]
There's a variable that should be exported in strongarm section named node_view_<SOMETHING> that enables the panel. Please ensure you export that variable.

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc (hunk 10 above)
Comment (robherba) [reply modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc:55]: Yes

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.info (hunk 3 above)
Comment (robherba) [reply modules/features/manatime_feature_tasks/manatime_feature_tasks.info]: ok

File: modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc (hunk 9 above)
Comment (robherba) [reply modules/features/manatime_feature_tasks/manatime_feature_tasks.views_default.inc:117]: I don't understand what are you saying. I did not generate this code, it was generated through the interface.

**Comment by kporras07** [comment]
Awesome! Just a couple minor changes.
In View config:
- Category: "Manatime"
- Display Name: Tasks Listing.

**Comment by robherba** [comment]
Changes has been applied @kporras07 !

**Comment by kporras07** [comment]
@rohernandez  :shipit:

**Pull request merged**